src/common/log/logHelper.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+/**
+ * Helper for the log utility.
+ */
+
+import * as util from "util";
+import {InternalError, InternalErrorLevel} from "../error/internalError";
+
+export enum LogLevel {
+    None = 0,
+    Error = 1,
+    Warning = 2,
+    Info = 3,
+    Debug = 4,
+    Trace = 5
+}
+
+export enum LogChannelType {
+    Console = 0,
+    OutputChannel = 1,
+    WritableStream = 2
+}
+
+export class LogHelper {
+    public static MESSAGE_TAG: string = "[vscode-react-native]";
+    public static INTERNAL_TAG: string = "[Internal]";
+    public static ERROR_TAG_FORMATSTRING: string = "[Error : %s] ";
+    public static WARN_TAG: string = "[Warning]";
+    private static ERROR_CODE_WIDTH: string = "0000";
+    private static LOG_LEVEL_NAME: string = "RN_LOG_LEVEL";
+
+    public static get logLevel(): LogLevel {
+        let valName: string = process.env[LogHelper.LOG_LEVEL_NAME];
+        return (<any> LogLevel)[valName];
+    }
+
+    public static set logLevel(level: LogLevel) {
+        if (!level) {
+            return;
+        }
+
+        // Set the process env value
+        process.env[LogHelper.LOG_LEVEL_NAME] = LogLevel[level];
+    }
+
+    /**
+     * Determines the type of the log channel (LogChannelType).
+     */
+    public static getLogChannelType(targetChannel: any): LogChannelType {
+        console.assert(!!targetChannel, "targetChannel is undefined");
+        if (typeof targetChannel.log === "function") {
```

Comment:
Although actually we probably don't want to change to instanceof checks, because if you do `targetChannel instanceof OutputChannel` then you'll have to import `OutputChannel` from `vscode` and that will break the out-of-process side of things.